### PR TITLE
Create youtube_downloader.py

### PR DESCRIPTION
**Pull request by s18k**
Python program to download a Youtube video as a .mp4 file

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by dhruvmanila** [comment]
@algorithms-keeper review

**Comment by cclauss** [comment]
web_programming/youtube_downloader.py:1:1: F401 'os' imported but unused
web_programming/youtube_downloader.py:11:5: F841 local variable 'filename' is assigned to but never used

**Comment by dhruvmanila** [comment]
Testing out whether the bot is ignoring to check doctest existence in `web_programming/` directory:

@algorithms-keeper review

**Comment by dhruvmanila** [comment]
Hmm, not working, will take a look in a few hours.

**Comment by dhruvmanila** [comment]
@algorithms-keeper review

File: web_programming/youtube_downloader.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import os
+from pathlib import Path
+
+from pytube import YouTube
+
+
+def download(requestedLink):
```
Comment (cclauss) [inline web_programming/youtube_downloader.py:7]:
```suggestion
def download(url: str) -> None:
```

**Comment by dhruvmanila** [comment]
Working!

File: web_programming/youtube_downloader.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    link = requestedLink
+    video = YouTube(link)
```
Comment (cclauss) [inline web_programming/youtube_downloader.py:9]:
```suggestion
    video = YouTube(url)
```

File: web_programming/youtube_downloader.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    stream = video.streams.get_highest_resolution()
+    filename = stream.title
```
Comment (cclauss) [inline web_programming/youtube_downloader.py:11]:
```suggestion
```

**Review by cclauss — CHANGES_REQUESTED** [review]

**Pull request closed without merging**